### PR TITLE
Add functionality for resetting robots

### LINKED ISSUE #1348: Create "Set robot to idle"-endpoint
**Describe the new feature you would like to see**
Endpoint that sets robot to available and current area to null. 

Consider also:
Stopping all ongoing missions and emptying the queue if there are

This would be good for getting back to a usable state if some bug occur

**Describe the solution you'd like**

**How will this feature affect the current Threat Model?**

### PR DESCRIPTION
**Pull request by mrica-equinor**
Closes #1348

### CODE REVIEW

File: backend/api/Controllers/RobotController.cs [hunk 1]
```diff
@@ -637,5 +639,86 @@ public async Task<ActionResult> ResumeMission([FromRoute] string robotId)
             return Ok(missionRun);
         }
 
+        /// <summary>
+        ///   Empies the mission queue for the robot, stops the ongoing mission and sets the robot to available. Current area is set to null
```
Comment (aestene) [inline backend/api/Controllers/RobotController.cs:643]:
```suggestion
        ///   Empties the mission queue for the robot, stops the ongoing mission, sets the robot to available and current area is set to null
```

File: backend/api/Controllers/RobotController.cs [hunk 2]
```diff
@@ -637,5 +639,86 @@ public async Task<ActionResult> ResumeMission([FromRoute] string robotId)
...
+
+            try { await robotService.UpdateCurrentArea(robot.Id, null); }
+            catch (RobotNotFoundException)
+            {
+                string errorMessage = $"Failed to set current area to null for robot with id {robotId} because the robot was not found";
+                logger.LogWarning("{Message}", errorMessage);
+                return NotFound(errorMessage);
+            }
+
+            missionSchedulingService.TriggerRobotAvailable(new RobotAvailableEventArgs(robot.Id));
```
Comment (Eddasol) [inline backend/api/Controllers/RobotController.cs:716]: Doesn't this happen by itself when all missions are cancelled and status becomes available? Do we risk it running twice now?

Comment (mrica-equinor) [reply backend/api/Controllers/RobotController.cs:716]: I am not sure, can have a check on this later today

Comment (mrica-equinor) [reply backend/api/Controllers/RobotController.cs]: You were right so I removed it

File: backend/api/Controllers/RobotController.cs [hunk 3]
```diff
@@ -637,5 +638,81 @@ public async Task<ActionResult> ResumeMission([FromRoute] string robotId)
...
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult> ResetRobot(
+            [FromRoute] string robotId
+        )
+        {
+            Console.WriteLine(robotId);
+            Console.WriteLine("oooooooooo....................");
```
Comment (andchiind) [inline backend/api/Controllers/RobotController.cs:661]: Don't think these are load-bearing printouts so they can probably be deleted

**Review by andchiind — APPROVED** [review]
Looks good otherwise. Make sure to remove the printouts before merging.

File: backend/api/Controllers/RobotController.cs (hunk 3 above)
Comment (mrica-equinor) [reply backend/api/Controllers/RobotController.cs]: Damn can't believe i forgot these

**Pull request merged**